src/main/java/org/antlr/intellij/plugin/configdialogs/ANTLRv4ProjectSettings.java

```diff
@@ -0,0 +1,49 @@
+package org.antlr.intellij.plugin.configdialogs;
+
+import com.intellij.openapi.options.SearchableConfigurable;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+
+public class ANTLRv4ProjectSettings implements SearchableConfigurable {
+
+    private ConfigANTLRPerGrammar configurationForm;
+
+    private final Project project;
+
+    public ANTLRv4ProjectSettings(@NotNull Project project) {
+        this.project = project;
+    }
+
+    @NotNull
+    @Override
+    public String getId() {
+        return "ANTLR4ProjectSettings";
+    }
+
+    @Nls(capitalization = Nls.Capitalization.Title)
+    @Override
+    public String getDisplayName() {
+        return "ANTLR4 Project Settings";
+    }
+
+    @Nullable
+    @Override
+    public JComponent createComponent() {
+        configurationForm = new ConfigANTLRPerGrammar(project, ANTLRv4GrammarProperties.PROJECT_SETTINGS_PREFIX);
```

Comment:
It is a very good idea to reuse this component, but we should find a way to hide the `Cancel` and `Save` buttons, because they are redundant with the other ones provided by default by IntelliJ:

<img width="749" alt="image" src="https://user-images.githubusercontent.com/281528/60463696-f8600280-9c4c-11e9-842d-01a0d28c6c7d.png">

Also, it would be nice to implement `isModified()` correctly so that the Apply button is only enabled when actual modifications are made. At the moment, the menu title appears in blue as if something was modified, except I didn't change anything:

<img width="227" alt="image" src="https://user-images.githubusercontent.com/281528/60463808-3bba7100-9c4d-11e9-9a7a-3325d0076b3b.png">

Reply:
Done